### PR TITLE
Renaming a parameter for Set-AzSKPIMConfiguration, bug fixes and PIM helper command enhancements

### PR DESCRIPTION
**Pull request by v-zhgup**
## Description
 Renaming a parameter for Set-AzSKPIMConfiguration and exception handling bug fix.
 Enhancements include:
1) Assigning PIM roles to groups 
2) Capability in Get-AzSKPIMConfiguration to list soon to expire PIM Assignments
3) Addition of a parameter in Set-AzSKPIMConfiguration to extend soon to expire PIM roles
4) Fix for API PIM changes and 'ExternalId' not found
5) Few comment addition
## Checklist
- [x] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [x] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [x] The title of the PR clearly describes the intent of the PR.
- [x] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK/Framework/Core/PIM/PIMScript.ps1 [hunk 1]
```diff
@@ -560,6 +541,90 @@ class PIM: AzCommandBase {
...
         }
     }
+
+    # Get the assignments that are expiring in n days
+    hidden  [PSObject] ListSoonToExpireAssignments($SubscriptionId, $ResourceGroupName, $ResourceName, $RoleNames, $ExpiringInDays)
+    {
+        $this.AcquireToken();
+        $criticalRoles = @();
+        $soonToExpireAssignments = @();
+        $criticalRoles += $this.ConvertToStringArray($RoleNames)
```
Comment (VishalHaibatpure) [inline src/AzSK/Framework/Core/PIM/PIMScript.ps1]: We can keep role name optional

File: src/AzSK/Framework/Core/PIM/PIMScript.ps1 [hunk 2]
```diff
@@ -560,6 +541,90 @@ class PIM: AzCommandBase {
...
+        $soonToExpireAssignments = $this.ListSoonToExpireAssignments($SubscriptionId, $ResourceGroupName, $ResourceName, $RoleNames, $ExpiringInDays);
+        $AssignmetntCount = ($soonToExpireAssignments | Measure-Object).Count
+        $ts =$DurationInDays
+        $url = $this.APIroot +"/roleAssignmentRequests "
+        if($AssignmetntCount -gt 0)
+        {
+            # If force switch is used extend to expire without any prompt
+            $this.PublishCustomMessage("");
+            $this.PublishCustomMessage("Initiating assignment extension for [$AssignmetntCount] PIM assignments..."); #TODO: Check the color
+            $UserResponse = 'N'
```
Comment (VishalHaibatpure) [inline src/AzSK/Framework/Core/PIM/PIMScript.ps1]: use readhost here

File: src/AzSK/PIM/PIM.ps1 [hunk 3]
```diff
@@ -87,19 +101,22 @@ function Set-AzSKPIMConfiguration {
...
+        [Alias("GroupName")]
         [ValidateNotNullOrEmpty()]
 	[Alias("pn")]
         [string]
         $PrincipalName,
 
+        
         [Parameter(Mandatory = $false, ParameterSetName = "RemovePermanentAssignment")]
         [ValidateNotNullOrEmpty()]
         [ValidateSet("MatchingEligibleAssignments", "AllExceptMe")]
```
Comment (VishalHaibatpure) [inline src/AzSK/PIM/PIM.ps1]: Add short alias

File: src/AzSK/Framework/Core/PIM/PIMScript.ps1 (hunk 1 above)
Comment (v-zhgup) [reply src/AzSK/Framework/Core/PIM/PIMScript.ps1]: will check and do as suggested

**Pull request merged**